src/client/clientmap.cpp

```diff
@@ -650,7 +650,23 @@ void ClientMap::renderMapShadows(video::IVideoDriver *driver,
 
 	MeshBufListList drawbufs;
 
+	int count = 0;
+	int low_bound = is_transparent_pass ? 0 : m_drawlist_shadow.size() / total_frames * frame;
+	int high_bound = is_transparent_pass ? m_drawlist_shadow.size() : m_drawlist_shadow.size() / total_frames * (frame + 1);
+
+	// transparent pass should be rendered in one go
+	if (is_transparent_pass && frame != total_frames - 1) {
```

Comment:
So we doing the transparent pass in the last frame? Why not the first frame? I.e. `if (is_transparent_pass && frame > 0) ...`

Reply:
This is because there is no double buffering for colored shadows, but they still render with 'future' frustum. The very last frame is exactly before 'future' becomes 'current' and I avoid visual glitches. I think I'll need to document this and other subtle things carefully.